bundles/org.openhab.binding.bluelink/src/main/java/org/openhab/binding/bluelink/internal/handler/BluelinkVehicleHandler.java

```diff
@@ -0,0 +1,496 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.bluelink.internal.api.BluelinkApi;
+import org.openhab.binding.bluelink.internal.api.BluelinkApiException;
+import org.openhab.binding.bluelink.internal.config.BluelinkVehicleConfiguration;
+import org.openhab.binding.bluelink.internal.dto.VehicleInfo;
+import org.openhab.binding.bluelink.internal.dto.VehicleStatus;
+import org.openhab.binding.bluelink.internal.dto.VehicleStatus.VehicleStatusData;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BluelinkVehicleHandler} handles a single Bluelink vehicle.
+ *
+ * @author Marcus Better - Initial contribution
+ */
+@NonNullByDefault
+public class BluelinkVehicleHandler extends BaseThingHandler {
+
+    private static final int DEFAULT_REFRESH_INTERVAL_MINUTES = 30;
+    private static final int DEFAULT_FORCE_REFRESH_INTERVAL_MINUTES = 240;
+
+    private final Logger logger = LoggerFactory.getLogger(BluelinkVehicleHandler.class);
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> forceRefreshJob;
+    private @Nullable VehicleInfo vehicleInfo;
+    private boolean isElectric = false;
+
+    public BluelinkVehicleHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(VehicleActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Bluelink vehicle handler");
+
+        final BluelinkVehicleConfiguration config = getConfigAs(BluelinkVehicleConfiguration.class);
+        final String vin = config.vin;
+        if (vin == null || vin.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "VIN is required");
```

Comment:
See other comment about i18n custom keys, also applies to occurences in this class.